### DISCUSSION TITLE
How to mock APIs with playwright?

### DISCUSSION THREAD

**Discussion by emnabn** [akshayp7/playwright-typescript-playwright-test#44] [General]
Hello,
I tried to mock an API in my project playwright with cucumber js but I didn't succeed:
the code that I implement in the step:

`  await page.route(
    '**/emobgapi/v1/offers?country-of-residence=FR&offset=0&limit=10&pickup-station-code=ORYT01&drop-off-station-code=ORYT01&pickup-date=2024-01-08T10:00&drop-off-date=2024-01-22T10:00&driver-age=26&apikey=sL7SEuCNdoBj8SilHttiSPRLnwpFSD92',
    route => {
      route.fulfill({
        status: 200,
        contentType: 'application/json',
        path: '~/tests/acceptance/searchAPI.json',
      });
    },
`

and I create `searchAPI.json` file where I put the response of this request.

Thanks for your help.

**Comment by akshayp7** [comment]
Hi, Thank you for showing interest in my work. Sorry I haven't worked with mocking API's much but surely I will look into this issue and get back if I fin anything

**Comment by akshayp7** [comment]
https://youtu.be/c7zCQ8FKih4
This might be useful
